CIP-0XXX/README.md

```diff
@@ -0,0 +1,561 @@
+|--------|------|------|
+| total | 1,148 B | 2,788 B | 
+
+For a total of **3,100** Cardano SPOs, there will be **1,550** messages sent per round of signature: 
+
+- the network outbound throughput of a peer is:
+
+| Send period | Lower bound | Upper bound |
+| ----------- | ----------- | ----------- |
+| 1 min       | 57 kB/s     | 138 kB/s    |
+| 2 min       | 29 kB/s     | 69 kB/s     |
+| 5 min       | 12 kB/s     | 28 kB/s     |
+| 10 min      | 6 kB/s      | 14 kB/s     |
+
+- the network outbound volume of a peer is:
+
+| Send period | Lower bound  | Upper bound  |
+| ----------- | ------------ | ------------ |
+| 1 min       | 147 GB/month | 356 GB/month |
+| 2 min       | 74 GB/month  | 178 GB/month |
+| 5 min       | 30 GB/month  | 72 GB/month  |
+| 10 min      | 15 GB/month  | 36 GB/month  |
+
+### Infrastructure extra operating costs
+
+#### Networking traffic cost
+
+> [!NOTE]  
+> - These data apply to cloud providers which bill the traffic on the volume, not the bandwidth.
+> - Some cloud providers offer a free tier for the first **100GB** of traffic which is not taken into consideration here for simplicity.
+
+| Cloud Provider | Inbound Traffic | Outbound Traffic |
+|--------|------|------|
+| AWS | 0 $/GB | 0.09 $/GB |
+| GCP | 0 $/GB | 0.11 $/GB |
+| Azure | 0 $/GB | 0.09 $/GB |
+| Average | 0 $/GB | 0.1 $/GB |
+
+
+#### Mithril message diffusion extra networking cost
+
+For a total of `3,000` SPOs sending messages, the extra networking cost incurred for a Cardano full node is:
+
+| Send period | Lower bound | Upper bound |
+| ----------- | ----------- | ----------- |
+| 1 min       | 15 $/month  | 36 $/month  |
+| 2 min       | 8 $/month   | 18 $/month  |
+| 5 min       | 3 $/month   | 8 $/month   |
+| 10 min      | 2 $/month   | 4 $/month   |
+
+### Protocol authentication
+
+#### Message authentication mechanism
+The message body is signed with the KES key of the SPO. This signature and the operational certificate of the SPO are appended to the message which is diffused.
+
+Before being diffused to other peers, an incoming message must be verified by the receiving node. This is done with the following steps:
+- Verify that the operational certificate is valid by checking that the KES verification key is signed by cold secret key.
+- Verify the KES signature of the message body with the KES verification key from the operational certificate.
+- Compute the SPO pool id by hashing the cold verification key from the operational certificate.
+- Make sure that this pool id is part of the stake distribution (the network layer will need to have access to this information).
+
+If any of these step fails, the message is considered as invalid, which is a protocol violation.
+
+> [!WARNING]  
+> We also probably need to make sure that the KES key used to sign is from the latest rotation:
+> - either the last seen opcert number in the block headers of the chain.
+> - or the last seen opcert number from a previous message diffused.
```

Comment:
Would the mithril signer registration (which needs to be decentralized) be a good source of truth here?

Reply:
Maybe in the long run, as the Mithril cryptography is not implemented in the node yet. Also this mechanism is supposed to work with future procotols which will probably only have access to KES keys and OpCert.